Codec/Archive/Tar/Index/Utils.hs

```diff
@@ -0,0 +1,41 @@
+module Codec.Archive.Tar.Index.Utils where
+
+import Data.ByteString as BS
+import Control.Exception (assert)
+
+import Data.ByteString.Internal (ByteString(..), unsafeWithForeignPtr, accursedUnutterablePerformIO)
+import GHC.Int (Int(..), Int32)
+import GHC.Word (Word32(..), byteSwap32)
+import Foreign.Storable (peek)
+import GHC.Ptr (castPtr, plusPtr, Ptr)
+import Data.Array.Base
+import Data.Array.Base
+import Data.Array.IO.Internals (unsafeFreezeIOUArray)
+
+-- | Construct a `UArray Word32 Word32` from a ByteString of 32bit big endian
+-- words.
+beToLe :: Word32
+       -- ^ The total array length (the number of 32bit words in the array)
+       -> BS.ByteString
+       -- ^ The bytestring from which the UArray is constructed.
+       -- The content must start in the first byte! (i.e. the meta-data words
+       -- that shouldn't be part of the array must have been dropped already)
+       -> UArray Word32 Word32
+beToLe lenArr (BS fptr _) = accursedUnutterablePerformIO $
+  unsafeWithForeignPtr fptr $ \ptr -> do
+    let ptr' = castPtr ptr :: Ptr Word32
+    unsafeFreezeIOUArray =<<
+      newGenArray (0, lenArr - 1) (\offset -> do
```

Comment:
I can see that some test data results in `beToLe` being called with `lenArr = 0 :: Word32`, in which case `lenArr - 1 == maxBound` and `peek` below results in segfault.

Reply:
At least now it builds 😄. I bumped the array min version to 0.5.6 to support `newGenArray`. Is that fine?

I'll look into the actual failure tomorrow.

Reply:
I suppose the solution here is to simply guard and `| lenArr == 0 = empty unboxed array`